lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -329,7 +340,14 @@ sub _schedule_iso {
         for my $job (@created_jobs) {
             $job->calculate_blocked_by;
         }
-        $gru->enqueue_download_jobs($downloads, \@successful_job_ids);
+        my %downloads = map {
+            $_ => [
+                scalar(keys %{$tmp_downloads{$_}->{destination}}) > 1
+                ? [keys %{$tmp_downloads{$_}->{destination}}]
+                : keys %{$tmp_downloads{$_}->{destination}},
```

Comment:
Looks like making the destinations always an array ref would also work and simplify things.